server/src/test/java/org/elasticsearch/common/util/concurrent/EsExecutorsTests.java

```diff
@@ -859,46 +856,68 @@ public void testScalingWithEmptyCoreAndKeepAliveAndWorkerPoolProbing() {
     }
 
     private void testScalingWithEmptyCore(EsThreadPoolExecutor executor) {
-        try {
-            class Task extends AbstractRunnable {
-                private int remaining;
-                private final CountDownLatch doneLatch;
-
-                Task(int iterations, CountDownLatch doneLatch) {
-                    this.remaining = iterations;
-                    this.doneLatch = doneLatch;
-                }
 
-                @Override
-                public void onFailure(Exception e) {
-                    fail(e);
-                }
+        class TaskScheduler extends SubscribableListener<Void> {
+            final ExecutorService scheduler;
+            final CyclicBarrier cyclicBarrier;
+            final Semaphore taskCompletions;
+            volatile int remaining;
```

Comment:
Does this need to be `volatile`? It's only modified when the barrier is tripping, so there's already a happens-before relationship with all of its reads. Or, to put this differently, I was suspicious by the `remaining--` on a bare volatile so either it needs a comment about its correctness or else we could save the next person reasoning this out fully by using an `AtomicInteger` with `decrementAndGet`.

Reply:
good point about the volatile, there's no need for it 👍